src/node-red/getting-started/node-red-messages.md

```diff
@@ -0,0 +1,236 @@
+---
+eleventyNavigation:
+  key: Node-RED Messages
+  parent: Getting Started
+meta:
+  title: Understanding Node-RED Messages
+  description: A practical guide to working with Node-RED messages, ensuring error-free flows and optimized data handling.
+  keywords: msg.payload node red, node red message structure, node-red messages, node red msg.payload, msg.payload, node red msg.payload array
+---
+
+# {{ meta.title }}
+
+Node-RED relies on passing messages between nodes to build dynamic IoT, automation, and data-processing workflows. Each message carries data that nodes modify, process, or analyze, so understanding how these messages are structured and handled is crucial. Mismanaging messages can lead to subtle bugs, such as overwriting data, infinite loops, or crashes. To prevent errors and build reliable flows, it’s essential to know how to work with messages efficiently.
+
+<!--more-->
+
+In this guide, we’ll explore the inner workings of Node-RED messages, common mistakes, and best practices for ensuring smooth data flow without errors.
+
+## What Are Node-RED Messages?
+
+In Node-RED, messages are packets of data that travel between nodes in your flow. Node-RED is event-driven, meaning an event emitter triggers events, and a listener reacts to them. In this context, nodes in Node-RED act as both event emitters and listeners, with messages serving as the medium through which these events are communicated. These messages represent the information that nodes read, modify, and act upon. Each node processes these messages and passes them along to the next node in the sequence.
+
+![Node-RED message passing animation](./images/node-red-message-passing.gif){data-zoomable}
+_Node-RED message passing animation_
+
+Messages are crucial because they carry the data that powers your workflows—whether it’s sensor readings, user inputs, or responses from an API. Essentially, Node-RED messages are JavaScript objects, which are simple yet powerful structures for managing and transferring data throughout your flows.
+
+### Understanding JavaScript Objects
+
+A [JavaScript object](https://www.youtube.com/watch?v=BRSg22VacUA) is a data structure used to store multiple values in a single variable. It consists of key-value pairs, where each key (or property) is associated with a specific value. This organization allows you to group and access related data easily.
+
+For example:
+
+```javascript
+const person = {
+  name: "Bob",
+  age: 24,
+  married: true
+};
```

Comment:
With the const it's no longer JSON, but a statement and allocation. Let's keep it easier to understand and just use JSON.
```suggestion
{
  name: "Bob",
  age: 24,
  married: true
}
```